v5.16.0 (#5902)
mui/mui-x: CHANGELOG.md

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 5.14.0

_Aug 25, 2022_

We'd like to offer a big thanks to the 11 contributors who made this release possible. Here are some highlights ✨:

- 🎁 Introduce column grouping for data grid (#5133) @alexfauquette

  You can now group columns using the `columnGroupingModel` prop. This lets you to display more structured data.

>   ![image](https://user-images.githubusercontent.com/45398769/186178366-4fba66b2-bf90-4c7a-9d83-940a7fc78704.png)

Reviewer: The screenshot is blurry at full width.

```suggestion
  <img src="https://user-images.githubusercontent.com/45398769/186178366-4fba66b2-bf90-4c7a-9d83-940a7fc78704.png" width="800" />
```

  To enable this feature, add `experimentalFeatures={{ columnGrouping: true }}`.
  The grouping header can be fully customized.
  See the [documentation](https://mui.com/x/react-data-grid/column-groups/) to explore everything it has to offer.

> - 🐞 Bugfixes

Reviewer 2: ```suggestion
- 🐞 Bugfixes

- 🌏 New locales thanks to @tesseractjh and @drastus
```

> - 🐞 Bugfixes

Reviewer 3: ```suggestion
- 🐞 Bugfixes
- 📚 Documentation improvements
- 🌏 New locales thanks to @tesseractjh and @drastus
```
🙈

> - 🧪 Add new unstable [`DateField`](https://mui.com/x/react-date-pickers/date-field/) and [`DateRangeField`](https://mui.com/x/react-date-pickers/date-range-field/) components for date pickers (#5504) @flaviendelangle

Author: @flaviendelangle is it a good time to include these components in the highlights? Or is it too early?

Reviewer 4: On one side they are to early stage to be used in a real application so we should not advertise about them

On the other side, maybe it's better to be explicit and say "Hey they exist, but be careful" and to start to gather feedback from very early testers (but if we have too many they will be tons of feedbacks like "Mobile is broken" which we know)

Reviewer 2: For now, we will get too many issues if users try to use it. Better to highlight this when we will have fix most of the issues we are already aware about.

Moreover we have no documentation, so they will struggle to use the component before realizing all the limitations

> - 🧪 Add new unstable [`DateField`](https://mui.com/x/react-date-pickers/date-field/) and [`DateRangeField`](https://mui.com/x/react-date-pickers/date-range-field/) components for date pickers (#5504) @flaviendelangle

Reviewer 4: ```suggestion
- 🧪 Add new unstable [`DateField`](https://mui.com/x/react-date-pickers/date-field/) and [`DateRangeField`](https://mui.com/x/react-date-pickers/date-range-field/) components for date pickers (#5504) @flaviendelangle

  ⚠️ Those components are very early stage and should not be used in production. The documentation will be published when they will be mature enough to be experimented by the community.
```

### `@mui/x-data-grid@v5.14.0` / `@mui/x-data-grid-pro@v5.14.0` / `@mui/x-data-grid-premium@v5.14.0`

#### Changes

- [DataGrid] Implement column grouping (#5133) @alexfauquette
- [DataGrid] Handle `disableVirtualization` prop change (#5889) @cherniavskii
- [DataGrid] Improve `GridRowModel` typing (#5734) @cherniavskii
- [DataGrid] Update deprecation note for `GridColDef` `hide` property (#5886) @cherniavskii

### `@mui/x-date-pickers@v5.0.0-beta.7` / `@mui/x-date-pickers-pro@v5.0.0-beta.7`

#### Changes

- [DatePicker] Fix to pass down `className` prop provided on DatePicker to `renderInput` (#5471) @CruseCtrl
- [DatePicker] Improve `a11y` support (#5809) @LukasTy
- [pickers] Add `PaperContent` component slot (#5801) @LukasTy
- [pickers] Add a breaking change section in the migration guide (#5805) @alexfauquette

> - [pickers] Add key in the `localeText` (#5143) @alexfauquette

Reviewer 4: ```suggestion
- [pickers] Add new translations to `localeText` (#5143) @alexfauquette
```

> - [pickers] Document components slots like for the data grid (#4657) @flaviendelangle

Reviewer: ```suggestion
- [pickers] Document components slots (#4657) @flaviendelangle
```

- [pickers] Add new unstable field components (#5504) @flaviendelangle
- [pickers] Fallback to default `minDate` / `maxDate` when `null` value is passed (#5397) @flaviendelangle
- [l10n] Add Korean (ko-KR) locale (#5854) @tesseractjh
- [l10n] Add Polish (pl-PL) locale (#5833) @drastus

### Docs

- [docs] Fix typo in `migration from lab` (#5277) @chuckwired
- [docs] Use `dayjs` instead of `date-fns` in doc examples (#5481) @flaviendelangle

### Core

> - [core] Clarify the scope of this license key (#5824) @oliviertassinari

Reviewer 4: ```suggestion
- [core] Clarify the scope of the license key used for tests and documentation (#5824) @oliviertassinari
```

> - [core] Clarify the scope of this license key (#5824) @oliviertassinari

Reviewer: ```suggestion
- [core] Clarify the scope of the license key (#5824) @oliviertassinari
```

- [core] Fix Typescript error on field hooks (#5892) @flaviendelangle
- [core] Memoize `columns` in `useDemoData` hook (#5848) @cherniavskii
- [core] Remove Firefox from the BrowserStack list (#5874) @DanailH

> - [core] Small changes to the release script (#5840) @m4theushw

Reviewer 3: Nitpick: what do you think about removing these from the changelog?
These are internal changes in no way related to what the release delivers... 🤔

Author: Historically we include everything in the release notes.
I have no strong preference on this, but sometimes it helps to find some clues what caused a regression
